internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -108,14 +112,17 @@ func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AServ
 }
 
 // GetTlsConfigurationForServer returns a tls.Config instance for use by a server application.
-func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient) (*tls.Config, error) {
+func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient, tokenManager tokenmanager.AccessTokenManager) (*tls.Config, error) {
```

Comment:
The logic on the server-side is a little more complicated (because of the SNI extension), but for now we should do the following: we need to accept an array of local identity pointers to GetTlsConfigurationForServer as well, and iterate through the list, for each pointer, if this pointer is non-nil then we call the tokenManager's Token(local identity) API, and if this pointer is nil then we call tokenManager.DefaultToken.

When we move the call to S2Av2 to a helper function in the follow-up PR, we will also need to add an extra auth mechanism for the SNI. Please add a TODO so we don't forget to do this. :)

Reply:
Thanks for the explanation! A question: what if there are > 1 non - nil local identities in the array? How do we know we've picked the right identity to get the corresponding token of? 

I'm assuming that once we integrate the SNI extension, we can use the SNI to pick with local identity from the list we would like to get the corresponding token for, but for now, will the array of local identities just be of size 1?

Reply:
> A question: what if there are > 1 non - nil local identities in the array? How do we know we've picked the right identity to get the corresponding token of?

We will include one AuthenticationMechanism for each in the array and the S2A will select which one to use.

> I'm assuming that once we integrate the SNI extension, we can use the SNI to pick with local identity from the list we would like to get the corresponding token for, but for now, will the array of local identities just be of size 1?

When we have the SNI, we will add one extra AuthenticationMechanism that has the SNI local identity and the corresponding token.

For now, yes the array will be of size 1.